detox/src/realms/DetoxPrimaryContext.test.js

```diff
@@ -1,62 +1,73 @@
-// @ts-nocheck
 const _ = require('lodash');
 
 const {
+  backupProcessEnv,
   latestInstanceOf,
   lastCallTo,
-  throwErrorImpl,
-  withSuspendingMock,
-  uuidRegexp,
-  tempFileRegexp,
-} = global;
-
-const workerId = 19;
+} = require('../../__tests__/helpers');
 
 describe('DetoxPrimaryContext', () => {
+  //#region Fixtures and constants
+  const UUID_REGEXP_STR = '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}';
+  const UUID_REGEXP = new RegExp(`^${UUID_REGEXP_STR}$`);
+  const TEMP_FILE_REGEXP = new RegExp(`.*${UUID_REGEXP_STR}.detox.json$`);
+  const WORKER_ID = 19;
+  const DETOX_CONFIG_BASE = Object.freeze({
+    behavior: {
+      init: {
+        keepLockFile: false,
+      }
+    },
+    device: {
+      type: '',
+    },
+    logger: {},
+    session: {
+      autoStart: false,
+    },
+  });
+  const FIRST_ARGUMENT = 0;
+  //#endregion
+
   let detoxConfigDriver;
 
+  //#region Mocks
+  /** @type {jest.Mocked<import('fs-extra')>} */
   let fs;
-  /** @type {import('../utils/logger')} */
+  /** @type {jest.Mocked<import('signal-exit')>} */
+  let signalExit;
+  /** @type {jest.Mocked<import('../logger')>} */
   let logger;
-
-  /** @type {import('../configuration')} */
+  /** @type {jest.Mocked<import('../configuration')>} */
   let configuration;
-  /** @type {import('../ipc/IPCServer')} */
+  /** @type {jest.Mock<import('../ipc/IPCServer')>} */
   let IPCServer;
-  /** @type {import('../ipc/IPCServer')} */
-  let ipcServer;
-
+  /** @type {jest.Mocked<import('../devices/lifecycle/GenyGlobalLifecycleHandler')>} */
   let globalLifecycleHandler;
-  /** @type {import('../environmentFactory')} */
+  /** @type {jest.Mocked<import('../environmentFactory')>} */
   let environmentFactory;
-  /** @type {import('../devices/DeviceRegistry')} */
+  /** @type {jest.Mocked<import('../devices/DeviceRegistry')>} */
   let deviceRegistryIOS;
-  /** @type {import('../devices/DeviceRegistry')} */
+  /** @type {jest.Mocked<import('../devices/DeviceRegistry')>} */
   let deviceRegistryAndroid;
-  /** @type {import('../devices/DeviceRegistry')} */
+  /** @type {jest.Mocked<import('../devices/DeviceRegistry')>} */
   let deviceRegistryGenyCloud;
-  /** @type {import('../server/DetoxServer')} */
+  /** @type {jest.Mock<import('../server/DetoxServer')>} */
   let DetoxServer;
+  /** @type {jest.Mock<import('../DetoxWorker')>} */
+  let DetoxWorker;
+  //#endregion
 
   /** @type {import('./DetoxInternalsFacade')} */
   let facade;
 
-  let env;
-
   const detoxServer = () => latestInstanceOf(DetoxServer);
-  const detoxWorker = () => {
-    const DetoxWorker = require('../DetoxWorker');
-    return latestInstanceOf(DetoxWorker);
-  };
+  const ipcServer = () => latestInstanceOf(IPCServer);
```

Comment:
After extracting all the types, these instance extractors turned into sleek one-liners.

Reply:
Well, it all was just about uniformity and readability – it's easier for the brain to scan over the boring same-pattern code than to read and try to understand why this place differs from the others.